### PR TITLE
Set correct owner and type for events in shared calendars, #fix3762

### LINKED ISSUE #3762: Missleading message in shared calendar
There is currently a missleading message when editing a calendar event in a shared calender. The message is "This event was sent from another calendar. Changes to the event will only be visible in your calendar". Which is not true in case of a shared calendar.

Steps to reproduce:

1. Invite another premium user to shared calendar with write permission.
2. Create event as participant in shared calendar
3. Edit event as owner 

The message does not show up  if:
* owner creates an event and participant edits the event
* participant creates the event and another particpant edits the event.
* the creator (participant/owner) of the event opens the event

**Test notes**

- [ ]  Invite another premium user to shared calendar with write permission, create event as participant in shared calendar, edit event as owner and observe that the message is not shown.
- [ ] Regression test permissions for calendar events (own, shared and invites).

### PR DESCRIPTION
**Pull request by sarashub**
For the owner of a shared calendar the type of an event that was created
by  another user was wrongfully set to 'invite'.
To prevent this behavior we set the organizer of such events to be the
owner of the calendar as only the owner should be able to add guests anyways.

close #3762

### CODE REVIEW

**Review by ganthern — CHANGES_REQUESTED** [review]
* [x] smoke tested
* [x] good commit message
* [ ] doc comments where necessary
* [ ] ~commits are squashed appropriately~
* [x] sufficient test notes
* [x] changes reviewed
* [ ] ~translation keys are in Phrase~

besides the code comments, the commit message title line seems broken (`#fix3762` instead of `fix #3762`, which should be in the last line of the message)

File: src/calendar/date/CalendarEventViewModel.ts [hunk 1]
```diff
@@ -520,6 +525,17 @@ export class CalendarEventViewModel {
...
+		//
+		// Case 1:
+		// Owner of the calendar created the event and invited some people. We, user with whom calendar was shared as RW, are seeing this event.
+		// We cannot modify that event even though we have RW permission because we are the not organizer.
+		// If the event is changed, the update must be sent out and we cannot do that because we are not the organizer.
+		//
+		// Case 2:
+		// Owner of the calendar received an invite and saved the event to the calendar. We, user with whom the calendar was shared as RW, are seeing this event.
+		// We can (theoretically) modify the event locally because we don't need to send any updates but we cannot change attendance because this would require sending an email.
+		// But we don't want to allow editing the event to make it more understandable for everyone.
```
[the remark this answers is not in the document]
Comment (charlag) [reply src/calendar/date/CalendarEventViewModel.ts:538]: Doc comments are about the interface mostly, this is about how it works, it's not useful to know when it happens for those who use it

File: src/calendar/date/CalendarEventViewModel.ts [hunk 2]
```diff
@@ -315,25 +300,45 @@ export class CalendarEventViewModel {
...
 	 *   ** this is not "our" copy of the event, from the point of organizer we saw it just accidentally.
 	 *   Later we might support proposing ourselves as attendee but currently organizer should be asked to
 	 *   send out the event.
 	 *
-	 *   *** depends on share capability. Cannot edit if it's not a copy and there are attendees.
+	 *   *** depends on share capability and whether there are attendees.
+	 *
+	 *   **** effectively organizer - if event doesn't have any guests then organizer doesn't matter and we can change
+	 *     the organizer to be ourselves. The only "real" organizer that event in a shared calendar can have is the
+	 *     owner of the calendar.
```
[the remark this answers is not in the document]
Comment (sarashub) [reply src/calendar/date/CalendarEventViewModel.ts:341]: There is a four-star note on organizer.

File: src/calendar/date/CalendarEventViewModel.ts [hunk 3]
```diff
@@ -315,46 +301,102 @@ export class CalendarEventViewModel {
 		}
 	}
 
+	private initOrganizer(existingOrganizer: EncryptedMailAddress | null, mailboxDetail: MailboxDetail, userController: IUserController): EncryptedMailAddress {
```
Comment (charlag) [inline src/calendar/date/CalendarEventViewModel.ts:304]: This can go now?

Comment (sarashub) [reply src/calendar/date/CalendarEventViewModel.ts:304]: yes!

**Review by charlag — APPROVED** [review]
LGTM!
There's a "guests" dropdown that doesn't do anything but I think it's not a big issue.

**Pull request merged**